environment.yml

```diff
@@ -1,8 +0,0 @@
-name: jupyterlab
```

Comment:
By adding a `binder/`, this file would no longer even be used. I would be :+1: keeping an annotated lab dev env as an `environment.yml` as a semi-canonical reference point... I don't really want to get into any holy war stuff about package managers, but i'll take the _get to `jlpm` on most platforms_ challenge with it. 

maybe something along the lines of:
```yaml
#   A JupyterLab development environment for Windows, Linux and MacOSX

name: jupyterlab-dev

#   Get started locally with the latest Miniconda Python 3.7:
#   - https://docs.conda.io/en/latest/miniconda.html

#   conda env update -p .env                # Install/update the env
#   conda activate -p .env                  # Activate the env
#   pip install -e .                        # Install source in local python
#   jupyter serverextension enable --py jupyterlab --sys-prefix  # serverextension
#   jlpm                                    # Install frontend build dependencies
#   jlpm build                              # Build the dev mode assets (optional)
#   jlpm build:core                         # Build the core mode assets (optional)
#   jupyter lab build                       # Build the app dir assets (optional)
#   jupyter lab --debug --dev-mode --watch  # Serve & watch/rebuild (optional)

channels:
- conda-forge

dependencies:
## Serving
- python >=3.7,<3.8
- notebook >=4.3.1
- jupyterlab_server >=0.2,<0.3

## Frontend Building
- nodejs >10,<11

## Testing
- pytest >=4.0,<5
- requests

## Documenting
# - sphinx
# - recommonmark
# - sphinx_rtd_theme
# - sphinx-auto-build
# - graphviz

## Used by jupyter-server-proxy
- aiohttp

## Not yet available from conda-forge
- pip:
  # Needed for binder, but you might not
  - jupyter-server-proxy==1.0beta9
  - simpervisor
  ## Docs
  # - pytest-check-links
```

Reply:
I would have no objection to adding a `binder/environment.yml` with roughly this in it. What do you think @yuvipanda? I would not necessarily want to add a `./environment.yml` if it was never built, since it would go stale pretty quickly...

Reply:
I generally very much prefer the minimum possible setup required to get this to work on binder inside the binder dir. Makes debugging and stuff a lot easier. I won't object to adding an environment.yaml file in the root of the repo, but that should maybe be a different issue / PR?

Reply:
Right, since this will _still_be a demo first, fast and simple is good. We're taking on a `jlpm && lab build` in the `postBuild`, which ain't free.

As to the staleness issue: throwing it on the CI matrix would keep it healthy: really, you just want to see "can i launch a non-template-failing `lab --watch`" on the three platforms....

Reply:
+1 to continuing the `environment.yml` discussion in a new PR.